test/shared/src/main/scala/zio/test/environment/TestClock.scala

```diff
@@ -95,7 +96,7 @@ import zio.clock.Clock
  */
 object TestClock extends Serializable {
 
-  trait Service extends Restorable {
+  trait Service extends Restorable with Clock.Service with Scheduler.Service {
```

Comment:
I noticed that the other test services don't extend the live services, which seems to create the possibility that the test service would not implement all the methods of the live service and prevent accessing the live service methods from the test service. Is there a reason for this? I am definitely new to this structure but I would have thought we would want to extend them as above.

Reply:
In the new scheme, covariance is not supported. That is to say, `Has[Clock.Service] with Has[TestClock.Service]` contains 2 modules, not 1, even if `TestClock.Service <: Clock.Service` (this is necessary for sane reasoning about module updates).

So it makes more sense to factor off the "test methods" from the "clock service", because the `TestClock.live` needs to emit _two_ things: `TestClock.Service` _as well as_ `Clock.Service`.

Does this make sense?

Basically it's an implementation detail that the same class below (`Test`) happens to implement both interfaces, it's not required though.